Sources/Rendering/Core/Picker/index.js

```diff
@@ -122,46 +284,33 @@ function vtkPicker(publicAPI, model) {
     if (selection.length !== 3) {
       vtkWarningMacro('vtkPicker::pick: selectionPt needs three components');
     }
+
+    if (!renderer) {
+      throw new Error('renderer cannot be null');
+    }
+
+    initialize();
+
     const selectionX = selection[0];
     const selectionY = selection[1];
     let selectionZ = selection[2];
-    let cameraPos = [];
-    let cameraFP = [];
-    let displayCoords = [];
-    let worldCoords = [];
-    const ray = [];
-    const cameraDOP = [];
-    let clipRange = [];
-    let tF;
-    let tB;
-    const p1World = [];
-    const p2World = [];
 
-    let props = [];
-    let pickable = false;
-    const p1Mapper = new Float64Array(4);
-    const p2Mapper = new Float64Array(4);
-    const bbox = vtkBoundingBox.newInstance();
-    const t = [];
-    const hitPosition = [];
-    const view = renderer.getRenderWindow().getViews()[0];
-
-    initialize();
     model.renderer = renderer;
     model.selectionPoint[0] = selectionX;
     model.selectionPoint[1] = selectionY;
     model.selectionPoint[2] = selectionZ;
 
-    if (!renderer) {
-      vtkErrorMacro('Picker::Pick Must specify renderer');
-      return;
-    }
+    let displayCoords = [];
+    const p1World = new Float64Array(4);
+    const p2World = new Float64Array(4);
```

Comment:
why mix Float64array and regular array ?

Reply:
To be honest, I didn't write those lines myself, only moved it around. It probably has a good reasons to use `Float64Array`. I'm going to use `Float64Array` everywhere since it makes it pre-allocates the memory and makes it clear how many items it contains.

Reply:
fair enough

Reply:
we compute `p1World` and `p2World`. on the other hand, we get `displayCoords` from another API